devel/validate-platforms

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+#
+# Validate different platform builds of the final Nextstrain image.
+#
+set -euo pipefail
+
+# Set default values.
+registry=localhost:5000
+tag=latest
+
+# Read command-line arguments.
+while getopts "r:t:" opt; do
+    case "$opt" in
+        r) registry="$OPTARG";;
+        t) tag="$OPTARG";;
+        *) echo "Usage: $0 [-r <registry>] [-t <tag>]" 1>&2; exit 1;;
+    esac
+done
+
+IMAGE="$registry/nextstrain/base:$tag"
+
+main() {
+    report_names=()
+
+    version_commands=(
+        "nextclade --version"
+        "augur --version"
+        "auspice --version"
+        "pip show evofr | grep Version"
+        "datasets --version"
+        "dataformat version"
```

Comment:
Ah, I don't think we want to ensure exact identity of third-party software versions between our platform images. This may not even be possible due to dependency resolution solutions varying between platforms. Even if it is, it'll be a continual headache, and not really the point of this image.

Reply:
Good to note the inconsistency here, but I also agree that it isn't necessary to validate against. Maybe it would be useful to print out the diff of `pip list` in case we want an easy way to look back at it on a particular tag. Is that what you were thinking @corneliusroemer?

Reply:
We definitely want to know about differences - there are not going to be many and they are very easy to check.

I agree we don't want to fail on these by default, but they're very useful to have available at a glance. This makes it _much_ easier to check whether there are differences and also just generally see which versions we have installed.

This would have helped debugging the bug that started all this _a lot_ (remember the biopython -> bcbiogff -> augur shebang).

While we may not want to fail for the entire dependency chain, I think it would be useful to ensure consistency of an important subset.

Wrote up a suggestion here: https://github.com/nextstrain/docker-base/pull/132

Reply:
Added small comments to #132. I was thinking something different but can be added independently of your proposal. Instead of failing on the diff of `pip list`, we can print it out separately to easily look back on the subtle differences such as `cryptography` package version differences between the two platform builds.

Reply:
Sure, either works. If we want to fail on few, we can just check in one style for simplicity. And make the full pip list pure info.